packages/compiler-cli/test/compliance/test_cases/r3_view_compiler_di/di/usefactory_with_deps.ts

```diff
@@ -1,8 +1,16 @@
-import {Injectable} from '@angular/core';
+import {Injectable, Optional} from '@angular/core';
 
 class SomeDep {}
 class MyAlternateService {}
 
 @Injectable({providedIn: 'root', useFactory: () => new MyAlternateService(), deps: [SomeDep]})
 export class MyService {
 }
+
+@Injectable({
+  providedIn: 'root',
+  useFactory: () => new MyAlternateService(),
```

Comment:
Although this is correct it doesn't really show the intention of using a dependency in the factory.
Perhaps this should be:

```suggestion
  useFactory: (dep: SomeDep|null) => new MyAlternateService(dep),
```

Reply:
Changed.